### PR TITLE
Java servlets

### PR DESCRIPTION
**Pull request by colleend**

### CODE REVIEW

File: java/servlets/security/cookie-issecure-false.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+rules:
+- id: cookie-issecure-false
+  patterns:
+  - pattern-either:
```
Comment (mschwager) [inline java/servlets/security/cookie-issecure-false.yaml]: Nit: we don't need this `pattern-either`. It makes it a little confusing as to which patterns are being `AND`'ed.

File: java/servlets/security/cookie-issecure-false.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        ...
+  - pattern-not-inside: |
+      $COOKIE = new Cookie(...);
+      ...
+      $COOKIE.setSecure(true);
+  message: |
+    Default session middleware settings: `setSecure` not set to true. 
+    This ensures that the cookie is sent only over HTTPS to prevent cross-site scripting attacks.
+  fix-regex:
+    regex: 'false'
```
Comment (mschwager) [inline java/servlets/security/cookie-issecure-false.yaml]: Should we be a little tighter on the bounds where `false` can occur? E.g. `\(false\)` -> `\(true\)`? My concern is that any `false` that occurs in the `pattern` via the ellipsis operator will be fixed, not necessarily just a `setSecure(false)`.

**Comment by mschwager** [comment]
Is this PR supposed to have 67 commits? I think you need to rebase on `develop`.

File: java/servlets/security/cookie-issecure-false.yaml (hunk 2 above)
Comment (colleend) [reply java/servlets/security/cookie-issecure-false.yaml]: good point -- I'll change this.

**Comment by colleend** [comment]
I've been having issues rebasing on develop 😱 but i'll try again after this

**Pull request closed without merging**